safe/impact_function/impact_function.py

```diff
@@ -312,6 +315,10 @@ def __eq__(self, other):
             'profiling',
         ]
         for if_property in properties:
+            # Skip if it's debug mode for profiling
+            if self.debug_mode:
+                if if_property == 'profiling':
```

Comment:
Use the layer purpose key from definition?

Reply:
Hmm, actually it's not a layer purpose key, but IF's attribute. So, I think I will keep using it.